File: src/OpenPeer.Domain/Entities/Comment.cs
namespace OpenPeer.Domain.Entities;

public class Comment
{
    public Guid Id { get; set; }
    public Guid PaperId { get; set; }
    public Guid UserId { get; set; }
    public Guid? ParentId { get; set; }
    public string Content { get; set; } = string.Empty;
    public bool IsDeleted { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    public Paper Paper { get; set; } = null!;
    public User User { get; set; } = null!;
    public Comment? Parent { get; set; }
    public ICollection<Comment> Replies { get; set; } = [];
}

File: src/OpenPeer.Web/src/types/comment.ts
export interface CommentDto {
  id: string;
  user: { id: string; userName: string; avatarPath?: string | null };
  content: string;
  parentId: string | null;
  replies: CommentDto[];
  createdAt: string;
  updatedAt: string | null;
}

export interface CreateCommentRequest {
  content: string;
  parentId?: string | null;
}

export interface UpdateCommentRequest {
  content: string;
}
